include/drivers/serial.h

```diff
@@ -148,6 +148,24 @@ union interrupt_enable_register {
 };
 typedef union interrupt_enable_register ier_t;
 
+union interrupt_identification_register {
+    uint8_t reg;
+    struct __packed {
+        uint8_t no_int_pend : 1, event : 3, rsvd : 1, fifo_64b : 1, fifo_status : 2;
+    };
+};
+typedef union interrupt_identification_register iir_t;
+
+#define UART_IIR_EVENT_MSR_CHANGE   0x0
+#define UART_IIR_EVENT_THR_EMPTY    0x1
+#define UART_IIR_EVENT_RXD_AVAIL    0x2
+#define UART_IIR_EVENT_LSR_CHANGE   0x3
+#define UART_IIR_EVENT_CHAR_TIMEOUT 0x6
+
+#define UART_IIR_FIFO_NO_FIFO       0x0
+#define UART_IIR_FIFO_UNUSABLE_FIFO 0x1
```

Comment:
I'm reading this: https://en.wikibooks.org/wiki/Serial_Programming/8250_UART_Programming#Interrupt_Identification_Register

Shouldn't this be 0x2?

> 


Bit 7 | Bit 6 | Notes
-- | -- | --
0 | 1 | Reserved condition
1 | 0 | FIFO enabled, but not functioning

Reply:
Hmm other sources seem to disagree:

http://www.larvierinehart.com/serial/serialadc/serial.htm#16

https://www.lammertbies.nl/comm/info/serial-uart

Reply:
Have you tested this or do we just trust either?